lib/facilities/ppms_client.rb

```diff
@@ -44,6 +44,26 @@ def provider_services(identifier)
       response.body
     end
 
+    def fetch_token
+      token = Redis.current.get('ppms-token')
+      return token if token
+      azure_client = Faraday.new("https://#{Settings.ppms.authority}")
+      auth_body = { 'encoding': 'utf8', 'api-version': '1.0',
+                    'client_id': Settings.ppms.client_id, 'client_secret': Settings.ppms.client_secret,
+                    'grant_type': 'client_credentials', 'resource': Settings.ppms.resource }
+      token_response = azure_client.post "/#{Settings.ppms.tenant}/oauth2/token", auth_body
+      token_hash = JSON.parse(token_response.body)
```

Comment:
... and if this had its own client implementation you can have Faraday middleware [parse the body](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/salesforce/configuration.rb#L13)